### ISSUE TITLE
Reply-to headers in emails not working

### ISSUE THREAD

**Issue by UnsolvedCypher** [thewca/worldcubeassociation.org#4601]
According to the code, there should be reply_to headers set on registration mailers (either to the organizer/delegates or to the competitor, depending on the specific email). However, when I look at registration emails I have received, these headers are missing. The code for the registration mailers is here: https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/mailers/registrations_mailer.rb

**Comment by viroulep** [comment]
Could you please tell use what mailer you're using?
As far as I could tell for the registrations emails I receive the `reply_to` headers were appropriately set (on gmail's web interface).
@danieljames-dj experienced something that may be related: when replying to an automatic competition results email, the `reply_to` header was not honored by the mailer (still gmail's web interface), but was appropriately set.

Sounds a bit different, but who knows...

**Comment by Jambrose777** [comment]
This just happened to me on a result posting:
![image](https://user-images.githubusercontent.com/11577241/65094731-1f94b900-d98c-11e9-9769-bfeacb2aa4a1.png)
In fact its happening for all mails of this type. I can also see it is now happening for all other results postings.

**Comment by UnsolvedCypher** [comment]
Sorry for the delay in responding to this. It looks like the headers are actually intact, but it appears that some clients ignored it. This recently happened for Gmail's web client, but some users have been using clients that ignore these headers even before that. Would it be possible to limit the emails that are sent from notifications@ and send them from another address instead? For example, could competition announcement confirmations be sent from competitons@ (the WCAT) and Delegate report submissions be sent from delegates@? This reduces how much the WCT inbox gets spammed by some of these requests.

**Comment by viroulep** [comment]
Yes, I think given this issue is not addressable by the WST, better choosing which address send which email sounds like a very good alternative!
Reports should definitely come from "reports@" though :)

If someone wants to look into our automatic email and which sender would be more appropriate, that would be awesome :)

**Comment by UnsolvedCypher** [comment]
Does the current SPF and/or DKIM setup allow us to send emails from any WCA domain email? If so, it might be helpful to send certain competition emails from the competition's Delegate rather than from notifications so that these emails make it to the Delegate rather than the WCT if the client ignores the reply_to headers.

**Comment by UnsolvedCypher** [comment]
Sending report/results emails has solved part of the problem, but the WCT still receives many responses to the registration pending/approved/deleted emails. Would it be appropriate to have these sent from the competition Delegate's email? I know it's not ideal, but it seems like a better approach than sending everything to the WCT and having them forward it. Alternatively, perhaps some unique code could be included in the original email that would correctly route it back to the organizers upon response?

**Comment by viroulep** [comment]
I don't think we want to be able to send email on behalf of everyone just because gmail can't handle reply-to-s properly.
I'd rather have these kind of emails sent from a "no-reply" email address if that's still a big pain to the WCT :s

>Alternatively, perhaps some unique code could be included in the original email that would correctly route it back to the organizers upon response?

Could you please give more details of what you're referring to?
I'm not sure I'm aware of something matching your description.

**Comment by UnsolvedCypher** [comment]
Sorry, I didn't describe that very well. One of the things that GitHub allows you to do is directly reply to an email in order to respond to a thread. Here are the headers on GitHub notification emails:

![Screenshot from 2019-11-10 15-10-27](https://user-images.githubusercontent.com/3170853/68549977-73d87a00-03cc-11ea-9c4c-5b6bf5ff8222.png)

It looks like what is going on is that emails are sent by notifications@github.com and then the `Message-ID` header is set to something that will identify the thread. When someone replies, the `In-Reply-To` header is set to the original message's ID by the user's client, which is used to figure out which thread the reply is for.

I think we could do something similar where we set the `Message-ID` to include the competition's ID, and based off of this we forward replies to the organizers automatically while still keeping notifications@ as the sender. We could also include this competition ID at the bottom of the email rather than using the `Message-ID` if that is preferred.

**Comment by Jambrose777** [comment]
@UnsolvedCypher does WCT still experience issues with this?

**Comment by dunkOnIT** [comment]
Closed as inactive
